File: LEETCODE/11. 盛最多水的容器.java
//⬇暴力
class Solution {
    public int maxArea(int[] height) {
        int ans = 0;
        int length = height.length;
        for (int i=0;i<length;i++) {
            for(int j=i+1;j<length;j++) {
                int area = Math.min(height[i], height[j]) * (j-i);
                ans = Math.max(area, ans);
            }
        }
        return ans;
    }
}

//⬇双指针
class Solution {
    public int maxArea(int[] height) {
        int len = height.length;
        int i = 0;
        int j = len - 1;
        int ans = 0;
        while ( i < j ){
            ans = Math.max(ans, Math.min(height[i], height[j]) * (j-i));
            if(height[i] <= height[j]) {
                i++;
            } else {
                j--;
            }
        }
        return ans;
    }
}

File: LEETCODE/kotlin/11. 盛最多水的容器.kt
import kotlin.math.min
import kotlin.math.max
class Solution {
    fun maxArea(height: IntArray): Int {
        var ans = 0
        val length = height.size
        for (i in height.indices) {
            for (j in i+1 until length) {
                var area = min(height[i], height[j]) * (j-i)
                ans = max(area, ans)
            }
        }
        return ans;
    }
}

class Solution {
    fun maxArea(height: IntArray): Int {
        val size = height.size
        var i = 0
        var j = size - 1
        var ans = 0
        while( i < j ){
            ans = max(ans, min(height[i], height[j])* (j-i)) 
            if (height[i] <= height[j]) i++ else j--
        }
        return ans
    }
}